### ISSUE TITLE
eslint incorrectly states unread variables are unused, even if they are written to.

### ISSUE THREAD

**Issue by mikemaccana** [eslint/eslint#10710]
**Tell us about your environment**

* **ESLint Version:** eslint@5.2.0
* **Node Version:** v10.7.0
* **npm Version:** 6.2.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

{
	"env": {
		"browser": true,
		"commonjs": true,
		"es6": true,
		"node": true,
		"mocha": true
	},
	"extends": "eslint:recommended",
	"parserOptions": {
		"ecmaVersion": 2017
	},
	"globals": {
		"mixpanel": true,
		"serverVars": true,
		"avkind": true,
		"kind": true,
		"Stripe": true,
		// Last two are from DigiCert site seal
		"__dcid": true,
		"__Cascade": true
	},
	"plugins": [
		"must-use-await"
	],
	"rules": {
		"indent": [
			"error",
			"tab",
			{"SwitchCase": 1}
		],
		"must-use-await/must-use-await": 1, // 1 warn, 2 error
		"linebreak-style": "off",
		"quotes": "off",
		"no-console": "off",
		"semi": "off",
		"no-unused-vars": "off",
		"no-debugger": "off"
	}
}
```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

	var doThings = function(){
		var x = 1
		x = 2
	}

	doThings;
```

For a more practical example:

```js
	var doThings = function(){
		var x = blah.find(findFunction)
		x = 2
		blah.save()
	}

```

```bash
eslint
```

**What did you expect to happen?**

Nothing. The variable is being written to and thus used.

**What actually happened? Please include the actual, raw output from ESLint.**

    'x is declared but its value is never read.'

**Comment by platinumazure** [comment]
Hi @mikemaccana, this is working as designed.

In this case, you could omit the assignment completely and nothing would break. If nobody is calling the function, why define it in the first place?

If this is supposed to be a global declaration, you need to use the `/* exported doThings */` comment syntax to tell ESLint that the variable is meant to be used in a different script. (Note that this only works if you're writing a script rather than a module.)

**Comment by not-an-aardvark** [comment]
Hi, thanks for creating an issue.

This is working as intended. As the error message says, `x` is assigned a value in your examples, but then that value is never read from `x`. So you could simply remove the `x` variable from your code without breaking anything.

**Comment by mikemaccana** [comment]
@platinumazure Someone is callign the function. `doThings()` is being run, and in the practical exampe, it's saving an object with the result ofthe modification, since the .find returns a reference to a value in `blah`.

**Comment by mikemaccana** [comment]
@not-an-aardvark I am aware x is never read. Thepoint is that x is written to. Removing x would mean that `blah` is not modified, as `x` refers to one of the values inside `blah`.

**Comment by mikemaccana** [comment]
@not-an-aardvark the value of `blah` returned by `find()` would not be set to `2` in this scenario

**Comment by not-an-aardvark** [comment]
The statement `x = 2` does not modify `blah` in your original code either, because it just reassigns the value of `x`. The value in `blah` will still be the same as it was before, even after running `x = 2`.

**Comment by mikemaccana** [comment]
Yeah you're right, I was just doing a reproduction and I realised that.

**Comment by mikemaccana** [comment]
PS. Thanks for your patience.
